src/dsymbol/string_interning.d

```diff
@@ -18,39 +18,80 @@
 
 module dsymbol.string_interning;
 
+import std.traits : Unqual;
 import dparse.lexer;
 
-/**
- * Interns the given string and returns the interned version.
- */
-istring internString(string s) nothrow @safe @nogc
+/// Obsolete, use `istring` constructor instead
+istring internString(string s) nothrow @nogc @safe
```

Comment:
use deprecated attribute

Reply:
Need to fix a lot of annoying messages in dcd, dscanner, and here. Is it worth doing?

Reply:
in this case I would prefer keeping this function as non-obsolete because it may arguably be better for readability, but on the other hand it could confuse over usage.

Reply:
But one doesn't see the return type. Using the constructor, it is clear:
```D
// how folks tend to write it
istring stuff = internString("stuff");
// but they can just
const stuff = istring("stuff");
```